src/components/cylc/workspace/Lumino.vue

```diff
@@ -140,71 +153,104 @@ onBeforeUnmount(() => {
-// eslint-disable-next-line no-unused-vars
-const closeAllViews = () => {
+async function closeAllViews () {
   for (const widget of Array.from(dockPanel.widgets())) {
     widget.close()
   }
+  await nextTick()
 }
 
 /**
- * Get the saved layout (if there is one) for the given workflow,
+ * Get the saved layout (if there is one) for the current workflow,
  * else add the default view.
- *
- * @param {string} workflowName
  */
-const getLayout = (workflowName) => {
-  restoreLayout(workflowName) || addView({ name: defaultView.value })
+async function getLayout () {
+  await restoreLayout() || await addView({ name: defaultView.value })
 }
 
 /**
- * Save the current layout/views to the store.
+ * Save the current layout/views to cache storage.
  */
-const saveLayout = () => {
-  $store.commit('app/saveLayout', {
-    workflowName: props.workflowName,
+async function saveLayout () {
+  // Serialize layout first to synchronously capture the current state
+  const serializedLayout = JSON.stringify({
     layout: dockPanel.saveLayout(),
-    views: new Map(views.value),
-  })
+    views: views.value,
+  }, replacer)
+  const cache = await layoutsCache
+  // Overrides on FIFO basis:
+  await cache.put(props.workflowName, new Response(
+    serializedLayout,
+    { headers: { 'Content-Type': 'application/json' } }
+  ))
+  const keys = await cache.keys()
+  if (keys.length > 100) {
+    await cache.delete(keys[0])
+  }
+}
+
+/**
+ * Return the saved layout for this workflow from cache storage, if it was saved.
+ */
+async function getStoredLayout () {
+  const cache = await layoutsCache
+  const stored = await cache.match(props.workflowName).then((r) => r?.text())
+  if (stored) {
+    return JSON.parse(
+      stored,
+      (key, value) => reviver(key, LuminoWidget.layoutReviver(key, value))
+    )
+  }
 }
 
 /**
- * Restore the layout for this workflow from the store, if it was saved.
+ * Restore the layout for this workflow, if it was saved.
  *
- * @param {string} workflowName
- * @returns {boolean} true if the layout was restored, false otherwise
+ * @returns true if the layout was restored, false otherwise
  */
-const restoreLayout = (workflowName) => {
-  const stored = $store.state.app.workspaceLayouts.get(workflowName)
+async function restoreLayout () {
+  const stored = await getStoredLayout()
   if (stored) {
+    layoutWatcher.pause()
     dockPanel.restoreLayout(stored.layout)
     // Wait for next tick so that Lumino has created the widget divs that the
     // views will be teleported into
-    nextTick(() => {
-      views.value = stored.views
-    })
+    await nextTick()
+    views.value = stored.views
+    layoutWatcher.resume()
     return true
   }
   return false
 }
 
+/**
+ * Reset the workspace layout to a single tab with the default view.
+ */
+async function resetToDefault () {
+  layoutWatcher.pause()
+  await closeAllViews()
+  await addView({ name: defaultView.value })
+  // (addView resumes the layout watcher)
+}
```

Comment:
:+1:

For future work, it would be great to allow the user to replace the "default view" with a "default layout" so we can allow users to save the current layout as their default view.